File: csharp/src/Structural/Decorator/TeaLeaves.cs
namespace Structural.Decorator {


//TeaLeaves.java - the decoratee

public class TeaLeaves : Tea {  
   public TeaLeaves() {
       teaIsSteeped = false;
   }
   
   override public string steepTea() {
       teaIsSteeped = true;
       return "tea leaves are steeping";
   }
}

}

File: scala/src/Structural/Decorator/TeaLeaves.scala
package Structural.Decorator;


//TeaLeaves - the decoratee

public class TeaLeaves extends Tea {  
   public TeaLeaves() {
       teaIsSteeped = false;
   }
   
   public String steepTea() {
       teaIsSteeped = true;
       return ("tea leaves are steeping");
   }
}
